### ISSUE TITLE
Cannot build static libraries on Windows

### ISSUE THREAD

**Issue by atilaneves** [atilaneves/reggae#73]
Including building a dub project that produces a static library.

**Comment by kinke** [comment]
Fixed by #114.

**Comment by aminya** [comment]
How can I use this package? There is no example of building a static library. The documentation is quite confusing.

I want to transform the following cmake file into reggae.d:
```cmake
cmake_minimum_required(VERSION 3.15)
project(mylib VERSION "0.1.0" LANGUAGES C)
set(CMAKE_STATIC_LIBRARY_PREFIX_C "")

# build mylib.lib
add_library(mylib STATIC "./src/lib.c")
target_include_directories(mylib PRIVATE "./lib/include")
set_target_properties(mylib PROPERTIES POSITION_INDEPENDENT_CODE ON)

```

**Comment by atilaneves** [comment]
> I have decoded the following from the docs, but it doesn't work:

When you say "it doesn't work" what exactly happens? What was the error message?

**Comment by aminya** [comment]
I get a lot of template instantiation errors. Could you add some copy-pastable examples for building static libraries?

**Comment by atilaneves** [comment]
Ah, I see what's wrong now. Well, one thing.

To make it possible for users to define targets and such at module-level, reggae's API uses a lot of templates to avoid trying to evaluate at compile-time that which only makes sense to evaluate at runtime. A problem in your sample was the `auto` instead of `alias`. 

I wrote a test for that though and quickly realised I broke `staticLibrary` ages ago and didn't notice because there were no tests for it... so I have a PR now to fix it.

The other problem I noticed with you example when I wrote it out is that sticking `"."` in `Sources` will make the rule pick up the hidden reggae source files, which is probably not what you wanted.
